pkg/sentry/platform/ring0/defs_amd64.go

```diff
@@ -72,19 +72,59 @@ type KernelOpts struct {
 	PageTables *pagetables.PageTables
 }
 
+// Currently, KVM_MAX_VCPUS is 288, which has been being increased slowly.
+// So 512 for _RING0_MAX_CPUS is enough for a while.
+const _RING0_MAX_CPUS = 512
+
+type cpuEntryArea struct {
+	alignStart [usermem.PageSize]byte
+
+	// cpuEntries is array of kernelEntry for all cpus
+	cpuEntries [_RING0_MAX_CPUS]kernelEntry
+
+	// globalIDT is our set of interrupt gates.
+	globalIDT idt64
```

Comment:
Why is this moved? I don't believe that it is actually required in this area since it is written once and never touched again.

Reply:
The cpu has to access to globalIDT in the upperhalf, so it must be moved together with the cpuEntries in the old patch. In new code, globalIDT is not moved, but it is also mapped in the upper area.